### PR TITLE
Contact Redirect

### LINKED ISSUE #4060: contact.php page redirection
when a user submits the contact form, rather than redirecting back to the contact form again, it should redirect them back to the forum index OR their last location prior to going to the contact page.

small UX improvement, I think..

### PR DESCRIPTION
**Pull request by effone**
Signed-off-by: effone <me@eff.one>

Resolves #4060

### CODE REVIEW

File: contact.php [hunk 1]
```diff
@@ -244,14 +244,15 @@
 			);
 			$db->insert_query("maillogs", $log_entry);
 		}
-
-		if($mybb->usergroup['emailfloodtime'] > 0 || (isset($sent_count) && $sent_count + 1 >= $mybb->usergroup['maxemails']))
+		
+		$mybb->input['from'] = $mybb->get_input('from');
+		if(isset($mybb->input['from']) && !empty($mybb->input['from']))
```
Comment (euantorano) [inline contact.php]: This can be simplified:

```suggestion
		if(!empty($mybb->input['from']))
```

**Review by euantorano — CHANGES_REQUESTED** [review]
Looks good, just a minor comment.

File: contact.php [hunk 2]
```diff
@@ -244,14 +244,15 @@
 			);
 			$db->insert_query("maillogs", $log_entry);
 		}
-
-		if($mybb->usergroup['emailfloodtime'] > 0 || (isset($sent_count) && $sent_count + 1 >= $mybb->usergroup['maxemails']))
```
Comment (JordanMussi) [inline contact.php:248]: Is this line intended to be lost? I don’t see how it relates to the ticket...

Comment (euantorano) [reply contact.php:248]: I hadn't even noticed that somehow.

Comment (effone) [reply contact.php:248]: Yes it is intended. That like was for specifically making a decision whether we can / should return to contact.php, on the contrary of the current scenario as we not at all wanna return on a successful submission.

Comment (effone) [reply contact.php:248]: ~~Wait wait, these conditions need to be placed somewhere, I need to revise it.~~

Comment (effone) [reply contact.php:248]: Well, I have rechecked it further. Email flood time as well as Sent limit is taken care in the beginning already.
This check is redundant and not applicable as we are not redirecting to `contact.php` further.
I believe what I have done here is correct. @JordanMussi @euantorano

**Comment by effone** [comment]
@euantorano please have a look on my last comment. I guess this PR is finished.

**Comment by euantorano** [comment]
> @euantorano please have a look on my last comment. I guess this PR is finished.

I'll have a look this afternoon and hopefully get this merged then, thanks!

**Comment by Eldenroot** [comment]
Works fine :)

**Comment by Eldenroot** [comment]
@euantorano - tested again, works fine.

File: install/resources/mybb_theme.xml [hunk 3]
```diff
@@ -14120,7 +14120,7 @@ $(function() {
 	{$sourcemode}
 });
 </script>]]></template>
-		<template name="contact" version="1816"><![CDATA[<html>
+		<template name="contact" version="1824"><![CDATA[<html>
```
Comment (euantorano) [inline install/resources/mybb_theme.xml]:
```suggestion
		<template name="contact" version="1825"><![CDATA[<html>
```

**Review by euantorano — APPROVED** [review]

**Pull request merged**